### ISSUE TITLE
LGTM.com - false positive: "Unused index variable" when using array to set order of execution

### ISSUE THREAD

**Issue by programmer2514** [github/codeql#9242]
**Description of the false positive**

I am creating a button toolbar, and I have an array called `buttonsOrder` that holds the position of each button to be inserted into the toolbar (say `[5,4,2,1,6,3,7]`), where the 1st button will be inserted 5th, the 2nd will be inserted 4th, the 3rd inserted 2nd, etc. To do this I am using the for loop below:

```
for (let i = 1; i <= buttonsOrder.length; i++) {
    if (i == buttonsOrder[0]) {
        // Insert 1st button
    }
    if (i == buttonsOrder[1]) {
        // Insert 2nd button
    }
    ...
}
```

This ensures the order of the buttons by only executing each index's corresponding code on its respective iteration, however, it is detected by LGTM as not using the index variable (due to me comparing `i` to the _value_ of each index, rather than using `i` _as_ an index). I am not aware of a better way to do what I am trying to do, although if it exists, please excuse my ignorance.

**URL to the alert on the project page on LGTM.com**

https://lgtm.com/projects/g/programmer2514/BetterDiscord-CollapsibleUI/snapshot/768904a74c3558d87e320c9b6618089032d8e627/files/CollapsibleUI.plugin.js?sort=name&dir=ASC&mode=heatmap#xc9cf67112902b80f:1

**Comment by edoardopirovano** [comment]
Indeed, this looks like a false positive. Thank you for reporting it!

Our current focus is on improving our security analysis. Because your report does not relate to a security query, we will put this on our backlog and prioritize it if we get enough reports of the same underlying issue in other projects. If you think that your report is related to our security analysis, please clarify that in a comment. Either way, we'll let you know here as soon as it's fixed!

As you might know, all of our queries are [open source](https://github.com/github/codeql). If you do have an idea for a code change, we encourage you to open a pull request. GitHub Code Scanning and lgtm.com have facilities for suppressing individual alerts or disabling a query.

cc. @github/codeql-javascript

**Comment by esbena** [comment]
JS analysis engineer here.

> I am not aware of a better way to do what I am trying to do

I think you are doing the right thing, it just happens to be similar enough to a very common mistake. I do not immediately see a good heuristic for whitelisting your use case. 

To suppress the alert you can [add an annotation manually](https://lgtm.com/help/lgtm/alert-suppression#javascript).

```javascript 
for (let i = 1; i <= buttonsOrder.length; i++) { // lgtm[js/unused-index-variable]
```
